src/smc-webapp/misc/clear-selection.ts

```diff
@@ -0,0 +1,24 @@
+/*
+ *  This file is part of CoCalc: Copyright © 2020 Sagemath, Inc.
+ *  License: AGPLv3 s.t. "Commons Clause" – see LICENSE.md for details
+ */
+
+// see http://stackoverflow.com/questions/3169786/clear-text-selection-with-javascript
+export function clear_selection() {
+  if (typeof window.getSelection === "function") {
+    if (window.getSelection().empty != null) {
+      // chrome
+      window.getSelection().empty();
+    } else if (window.getSelection().removeAllRanges != null) {
+      //firefox
+      window.getSelection().removeAllRanges();
+    }
+  } else {
+    if (
```

Comment:
Aren't these 6 lines the same as the one line `document.selection?.empty()`, since it has to be a function if defined in any modern browser.

In any case this approach (refactoring misc_page.coffee) should probably be done properly for the whole module, not just taking one random function out as a band-aide...